textattack/search_methods/metropolis_hastings.py

```diff
@@ -0,0 +1,175 @@
+            del input_ids
+        
+        perplexity = math.exp(loss)
+        return 1/perplexity
+        
+    def _stationary_dist(self, x, original_label):
+        """
+        Unnormalized estimation of the distribution that we want to sample from.
+        Args:
+            x (TokenizedText)
+            original_label
+        Returns: lm_score(x) * prob_model(wrong|x)
+        """
+        model_result = self.get_goal_results([x], original_label)[0][0]
+        lm_score = self._lm_score(x.text)
+        return lm_score * model_result.score
+
+    def _batch_stationary_dist(self, x_list, original_label):
+        """
+        Unnormalized estimation of the distribution that we want to sample from.
+        Process items in batch.
+        Args:
+            x_list (list): List of TokenizedText
+            original_label
+        Returns: list of floats representing lm_score(x) * prob_model(wrong|x)
+        """
+        batch_size = len(x_list)
+        model_results = self.get_goal_results(x_list, original_label)[0]
+        lm_scores = [self._lm_score(x.text) for x in x_list]
+        scores = [lm_scores[i] * model_results[i].score for i in range(batch_size)]
+
+        return scores, model_results
+
+    def _normalize(self, values):
+        """
+        Take list of values and normalize it into a probability distribution
+        """
+        s = sum(values)
+        if s == 0:
+            return [1/len(values) for v in values]
+        else:
+            return [v/s for v in values]
+
+    def _perform_search(self, initial_result):
+        text_len = len(initial_result.tokenized_text.words)
+        max_iter = max(self.max_iter, text_len)
+
+        current_result = initial_result
+        current_text = initial_result.tokenized_text
+        current_score = self._stationary_dist(current_text, initial_result.output)
+        
+        for n in range(max_iter):
+            # i-th word we want to transform
+            i = n % text_len
+
+            transformations = self.get_transformations(
+                        current_text, 
+                        indices_to_modify=[i],
+                        original_text=initial_result.tokenized_text
+                    )
+
+            if len(transformations) == 0:                
+                continue
+
+            scores, model_results = self._batch_stationary_dist(transformations, initial_result.output) 
+            proposal_dist = self._normalize(scores)
+            # Choose one transformation randomly according to proposal distribution
+            jump = np.random.choice(list(range(len(transformations))), p=proposal_dist)
+
+            # Now we have calculate probability of return proposal g(x'|x)
+            reverse_transformations = self.get_transformations(
+                        transformations[jump], indices_to_modify=[i],
+                        original_text=initial_result.tokenized_text
+                    )
+
+            reverse_jump = -1
```

Comment:
nit: use `None` and `is None` instead of `-1` and `==-1`

Reply:
done